hazelcast/src/main/java/com/hazelcast/internal/networking/ChannelCloseListener.java

```diff
@@ -16,13 +16,12 @@
 
 package com.hazelcast.internal.networking;
 
-import com.hazelcast.nio.Connection;
-
-public interface ChannelConnection extends Connection {
-
-    ChannelReader getChannelReader();
-
-    ChannelWriter getChannelWriter();
+/**
+ * A listener called when a {@link Channel} is closed.
+ *
+ * One of the potential usages is to release resources attached to a channel e.g. deregistration of metrics.
+ */
+public interface ChannelCloseListener {
 
-    Channel getChannel();
+    void onClose();
```

Comment:
Shouldn't a listener need to know the channel being closed? `void onClose(Channel channel)`